providers/shared/components/cache/id.ftl

```diff
@@ -49,6 +49,32 @@
                     }
                 ]
             },
+            {
+                "Names" : "MaintenanceWindow",
+                "Children" : [
+                    {
+                        "Names" : "DayOfTheWeek",
+                        "Types" : STRING_TYPE,
+                        "Values" : [
+                            "Sunday",
+                            "Monday",
+                            "Tuesday",
+                            "Wednesday",
+                            "Thursday",
+                            "Friday",
+                            "Saturday"
+                        ]
+                    },
+                    {
+                        "Names" : "TimeOfDay",
+                        "Types" : STRING_TYPE
+                    },
+                    {
+                        "Names" : "TimeZone",
+                        "Types" : STRING_TYPE
```

Comment:
Should we default the TimeZone to UTC to make it easier to follow?

Reply:
Maybe I configured it incorrectly, but if I set a default on the children o, I was then explicitly setting the maintenance window and couldn't leave it to AWS to configure an "appropriate" window (comment mostly applies to RDS which use the region to determine "appropriate")

Reply:
Thats the isConfigured check on the Maintenance window right? I.e. if someone has configured a maintenance window the defaut TimeZone is to use UTC.

Reply:
I don't see a function isConfigured. Is it in master? I have a isLayerConfigured and a isConfigurationIncluded but neither seems to be what you're talking about

Also I didn't think there was a difference between something set in a solution file, inherited from a shared solution file or set by a default, when it came to processing. Am I missing something?

Reply:
IsConfigured is added at run-time to say whether the parent has explicitly been defined. Do a global search in the aws provider and you'll see the usage.

Reply:
The function is isPresent and the property is .Configured (had a chat with Roley at the standup). I'm leaning towards .Configured, because isPresent relies on an Enabled:true as well and you can't disable maintenance windows, so I think that'd be misleading

The more I think about this, the more I'm in favour of making the day, time and timezone mandatory with no defaults rather than having someone specify to do maintenance on Wednesday at 01:00 and have it trigger at 11:00 due to not specifying the timezone and picking up the UTC default. Thoughts?

If a child attribute has Mandatory:true does that force the existence of the parent, or is the child attribute only mandatory if the parent is defined? ie Can I enforce using Mandatory:true or do I need to explicitly code a fatal error for a partial maintenance window specification?

Reply:
Mandatory applies to the whole configuration object not just a child configuration. So if the property is conditional then you need to handle it outside of the configuration process

Reply:
Cool, added standard function testMaintenanceWindow to handle including Azure v AWS options based on current preview